src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -143,36 +144,93 @@ private void ImportCalli(int token)
 
         private void ImportLoadNull()
         {
-            _interpreter.EvaluationStack.Push(new ObjectRefStackItem(null));
+            _interpreter.EvaluationStack.Push(StackItem.FromObjectRef(null));
         }
 
         private void ImportReturn()
         {
-            bool hasReturnValue = _interpreter.EvaluationStack.TryPop(out StackItem stackItem);
-            if (!hasReturnValue)
+            bool hasStackItem = _interpreter.EvaluationStack.TryPop(out StackItem stackItem);
```

Comment:
It would be better to check ReturnType first:

- If it is void, there is nothing to pop
- If it is not void, there has to be item to pop and not having one is InvalidProgramException. You may want to create a method that pops the item from the stack; and if there is not one, it throws InvalidProgramException. It will be the common method to use to work with the IL stack.